rules/CodeQuality/Rector/ClassConstFetch/ConvertStaticPrivateConstantToSelfRector.php

```diff
@@ -46,6 +64,38 @@ public function run()
 }
 CODE_SAMPLE
                     ,
+                    [
+                        self::ENABLE_FOR_NON_FINAL_CLASSES => false,
+                    ],
+                ),
+                new ConfiguredCodeSample(
+                    <<<'CODE_SAMPLE'
+class Foo {
+    private const BAR = 'bar';
+    public function run()
+    {
+        $bar = static::BAR;
+    }
+}
+class Bar extends Foo {
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class Foo {
+    private const BAR = 'bar';
+    public function run()
+    {
+        $bar = self::BAR;
+    }
+}
+class Bar extends Foo {
+}
+CODE_SAMPLE
+                    ,
+                    [
+                        self::ENABLE_FOR_NON_FINAL_CLASSES => true,
+                    ],
```

Comment:
please only define ConfiguredCodeSample once, only above, to avoid double/duplicated regeneration in docs

Reply:
Done, but how do you show what the options are doing otherwise?

Reply:
named configuration already show it

Reply:
Maybe not directly related to this PR, but wouldn't it be awesome to have multiple examples per rector with the different config options like [here](https://mlocati.github.io/php-cs-fixer-configurator/#version:3.13|fixer:concat_space)?

The named configuration works for you and me because I wrote the code and you reviewed that, but I fear it might not be that obvious if anyone is quickly skimming over all rules.

Reply:
that's probably a feature for docs generation needed, multiple example generation currently seems cause ambiguous wiht current generation. Feel free to provide a feature PR for docs generation.

Reply:
What do you think about it @TomasVotruba ?